local_db/data_access.py

```diff
@@ -425,3 +426,30 @@ def group_comment(
             )
 
             self._create_audit_log(audit)
+
+    def group_comment_delete(
+        self,
+        request_id: str,
+        group: str,
+        comment_id: str,
+        username: str,
+        audit: AuditEvent,
+    ):
+        with transaction.atomic():
+            # we can just get the comment directly by id
+            comment = FileGroupComment.objects.get(
+                id=comment_id,
+            )
+            # but let's verify we're looking at the right thing
+            if not (
+                comment.author == username
+                and comment.filegroup.name == group
+                and comment.filegroup.request.id == request_id
+            ):
+                raise BusinessLogicLayer.APIException(
+                    "Comment for deletion has inconsistent attributes "
+                    f"(in file group '{comment.filegroup.name}')"
+                )
```

Comment:
nb. maybe we don't need to pass these params? (& could then remove a test or two?)